src/react/components/common/tagInput/tagInputItem.tsx

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import { ITag } from "../../../../models/applicationState";
+import { constants } from "../../../../common/constants";
+// tslint:disable-next-line:no-var-requires
+const tagColors = require("../tagColors.json");
+
+export enum TagEditMode {
+    Color = "color",
+    Name = "name",
+}
+
+export interface ITagInputItemProps {
+    tag: ITag;
+    index: number;
+    isBeingEdited: boolean;
+    isLocked: boolean;
+    isSelected: boolean;
+    appliedToSelectedRegions: boolean;
+    onClick: (tag: ITag, ctrlKey: boolean, altKey: boolean, clickedColor?: boolean) => void;
+    onChange: (oldTag: ITag, newTag: ITag) => void;
+}
+
+export interface ITagInputItemState {
+    isBeingEdited: boolean;
+    isLocked: boolean;
+    tagEditMode: TagEditMode;
+    preventSingleClick: boolean;
+}
+
+const delay = 200;
+
+export default class TagInputItem extends React.Component<ITagInputItemProps, ITagInputItemState> {
+
+    public state = {
+        isBeingEdited: false,
+        isLocked: false,
+        tagEditMode: null,
+        preventSingleClick: false,
+    };
+
+    public render() {
+        const style: any = {
+            background: this.props.tag.color,
+        };
+        if (this.props.appliedToSelectedRegions) {
+            style.borderColor = this.props.tag.color;
+        }
+        return (
+            <div className={"tag-item-block"}>
+                {
+                    this.props.tag &&
```

Comment:
Is there a use case when we don't have a tag ?

Reply:
There were some instances where the tag was null during the rendering process and caused an error to be thrown